File: 02. Programming - Java/Lesson6/src/uni/fmi/bachelors/Student.java
package uni.fmi.bachelors;

public class Student extends Person{

	private String fn;
	private double averageGrade;
	private String specialty;
	
	public Student(String name, String egn) {
		super(name, egn);
		
	}

	public void calculateAverageGrade(int[] grades) {
		int total=0;
		for(int grade:grades) {
		total+=grade;
		}
		averageGrade=(double)total/grades.length;
		
	}
	
	@Override
	public String toString() {
		return "Student [fn=" + fn + ", averageGrade=" + averageGrade + ", specialty=" + specialty + "]"+super.toString();
	} 
	
}

File: 03. Programming - C#/TestPreparation3Example/Student.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TestPreparation3Example
{
    class Student:Person
    {
        private double averageGrade;
        public double AverageGrade
        {
            get
            {
                return averageGrade;
            }
            set
            {
                averageGrade = value;
            }
        }

        public Student(string name, string egn, string gender, double averageGrade)
            : base(name, egn, gender)
        {
            this.averageGrade = averageGrade;
        }
    }
}
